warehouse/query-core/src/main/java/datawave/query/transformer/ContentQueryTransformer.java

```diff
@@ -15,20 +15,92 @@
 import org.apache.accumulo.core.data.Key;
 import org.apache.accumulo.core.data.Value;
 import org.apache.accumulo.core.security.Authorizations;
+import org.apache.log4j.Logger;
 
 import java.util.ArrayList;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
 import java.util.Map.Entry;
 
 public class ContentQueryTransformer extends BaseQueryLogicTransformer<Entry<Key,Value>,EventBase> {
     
+    private static final Logger log = Logger.getLogger(ContentQueryTransformer.class);
+    
     protected final Authorizations auths;
     protected final ResponseObjectFactory responseObjectFactory;
+    protected final Map<Metadata,String> metadataIdMap;
     
     public ContentQueryTransformer(Query query, MarkingFunctions markingFunctions, ResponseObjectFactory responseObjectFactory) {
         super(markingFunctions);
         this.auths = new Authorizations(query.getQueryAuthorizations().split(","));
         this.responseObjectFactory = responseObjectFactory;
+        this.metadataIdMap = extractMetadadaIdMap(query);
+    }
+    
+    /**
+     * Extract optional identifiers from the query. Expected query format is:
+     * <p>
+     * 'DOCUMENT:shard/datatype/uid!optionalIdentifier1 DOCUMENT:shard/datatype/uid!optionalIdentifier2 .. DOCUMENT:shard/datatype/uid!optionalIdentifier3'
+     * <p>
+     * The identifiers are not required, so this will parse 'DOCUMENT:shard/datatype/uid' as well.
+     *
+     * @param querySettings
+     *            the current query for which we are transforming results.
+     * @return a map of shard/datatye/uid mapped to their corresponding identifiers.
```

Comment:
This is great documentation, it should also go in the query_help doc.

Reply:
Good point. I've been meaning to chase down the location of that.